src/components/application_manager/src/commands/request_from_hmi.cc

```diff
@@ -101,6 +101,28 @@ void RequestFromHMI::SendErrorResponse(
   rpc_service_.ManageHMICommand(message);
 }
 
+uint32_t RequestFromHMI::HMIAppId() {
+  smart_objects::SmartObject& so = *message_;
+  if (!so.keyExists(strings::msg_params)) {
+    LOG4CXX_DEBUG(logger_,
+                  strings::msg_params << " section is absent in the message.");
+    return 0;
+  }
+  if (!so[strings::msg_params].keyExists(strings::application)) {
+    LOG4CXX_DEBUG(logger_,
+                  strings::application << " section is absent in the message.");
+    return 0;
+  }
+  if (so[strings::msg_params][strings::application].keyExists(
+          strings::app_id)) {
+    LOG4CXX_DEBUG(logger_,
+                  strings::app_id << " section is absent in the message.");
+    return 0;
+  }
+  return static_cast<uint32_t>(
+      so[strings::msg_params][strings::application][strings::app_id].asUInt());
```

Comment:
@IGapchuk please make this function more effective. You getting `so[strings::msg_params]` 4 times here! 
You can use temporary variables (const ref) to access already checked internals of so